src/scenic/simulators/webots/model.scenic

```diff
@@ -54,25 +64,41 @@ class WebotsObject:
         positionOffset (`Vector`): Offset to add when computing the object's
             position in Webots; for objects whose Webots ``translation`` field
             is not aligned with the center of the object.
-        rotationOffset (float): Offset to add when computing the object's
-            rotation in Webots; for objects whose front is not aligned with the
+        rotationOffset (tuple[float, float, float]): Offset to add when computing the object's
+            orientation in Webots; for objects whose front is not aligned with the
             Webots North axis.
+        density (float): Density of this object. The corresponding Webots object
+            must have the ``density`` field.
 
     .. _Supervisor API: https://www.cyberbotics.com/doc/reference/supervisor?tab-language=python
     """
-
-    elevation[dynamic]: None
+    elevation[dynamic]: None if is2DMode() else float(self.position.z)
     requireVisible: False
 
+    webotsAdhoc: False
+
     webotsName: None
     webotsType: None
     webotsObject: None
 
     controller: None
     resetController: True
 
-    positionOffset: (0, 0)
-    rotationOffset: 0
+    positionOffset: (0, 0, 0)
+    rotationOffset: (0, 0, 0)
+
+    density[dynamic]: None # kg/m^3
```

Comment:
Can the density of an object really change during a simulation?

Reply:
This is so we can pull the density from Webots if it's not provided. Is there a cleaner way to do that?

Reply:
I was thinking we could do it `WebotsSimulation.__init__`, similar to how we handle `elevation`. But taking a look it seems that happens before setting the dynamic proxy for the object, which means we mutate the original object: i.e., if `elevation` is originally `None`, after the simulation it won't be any more, and the new value will leak into future uses of the scenario. If you can confirm this behavior, it's a bug, so we'd better figure out a different way to do this.

Reply:
Handled by your PR